### ISSUE TITLE
TridiagonalMatrix::compute_eigenvalues() does not throw

### ISSUE THREAD

**Issue by nfehn** [dealii/dealii#14609]
I want to try-catch a Newton-Krylov solver and adjust some parameters in case no convergence is achieved. The Krylov solver  (e.g. CG, FGMRES) uses geometric multigrid with Chebyshev smoothing as a preconditioner. I am using a matrix-free implementation.

The problem now is that `TridiagonalMatrix::compute_eigenvalues()` (called during initialization of the Chebyshev smoother) contains the line

https://github.com/dealii/dealii/blob/85f093567b355e1f8173598f2a479593a0506630/source/lac/tridiagonal_matrix.cc#L236

```C++
Assert(info == 0, ExcInternalError());
```

which fails in my case and which I can not catch.

Can we change this line to `AssertThrow` or where should this problem be handled?

As a sidenote, I could imagine this problem potentially raises robustness issues in the code related to the work https://onlinelibrary.wiley.com/doi/full/10.1002/nme.6336 for large deformations/displacements and high polynomial degrees, since I am considering a very similar problem of nonlinear structural mechanics (in the project ExaDG, https://github.com/exadg/exadg).

**Comment by kronbichler** [comment]
I fully agree, this is a run-time error and one should be able to catch it. Do you want to open a pull request that changes this line?
